### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by martinacantaro**
# Update readme

## Description

Goal: Improving documentation to provide useful info and make the experience of building this repo better. 
Related to: https://github.com/lambdaclass/cairo-rs/issues/713

Changes made:
- Provide context about the Cairo VM, Cairo Language and Starknet.
- Explain why and how to run code using the original VM
- Specify versions of dependencies
- Discriminate between required and optional dependencies.
- Improve the step by step description of building this repo and running a program.
- Now with emojis!

## Checklist
- [x] Linked to Github Issue
- [-] Unit tests added
- [-] Integration tests added.
- [-] This change requires new documentation.
  - [x] Documentation has been added/updated.

### CODE REVIEW

**Review by Juan-M-V — APPROVED** [review]

**Review by Jrigada — APPROVED** [review]
Awesome work, I would move the TODO section here to GH issues and do it in another PR.

File: README.md [hunk 1]
```diff
@@ -58,6 +90,14 @@ cairo-compile cairo_programs/abs_value_array.cairo --output cairo_programs/abs_v
 
 target/release/cairo-rs-run cairo_programs/abs_value_array_compiled.json --layout all
 ```
+### Using hints
+
+Currently, as this VM is under construction, it's missing some of the features of the original VM. Notably, this VM only implements a limited number of hints at the moment, while the [Python Cairo VM](https://github.com/starkware-libs/cairo-lang) allows for the implementation of user-defined hints.
```
Comment (fmoletta) [inline README.md:95]: "allows for the implementation of user-defined hints." This line is a bit confusing. As the original VM allows users to run user-defined hints (any python code), not having to implement them. Implementing the hints is something we do from our side to support a similar behaviour to the original VM. Maybe we could say "allows running any hint/any python code". instead

Comment (martinacantaro) [reply README.md]: Solved!

File: README.md [hunk 2]
```diff
@@ -25,28 +25,60 @@
...
-- [Rust](https://www.rust-lang.org/tools/install)
+**Required**
+- [Rust 1.66.1](https://www.rust-lang.org/tools/install)
 - Cargo
-- PyEnv for running the original VM and compiling cairo programs
+
+**Optional**
+These dependencies are only necessary in order to run the original VM and compile Cairo programs.
+- PyEnv with Python 3.9 
 - cairo-lang (optional)
```
[the remark this answers is not in the document]
Comment (martinacantaro) [reply README.md]: Solved!

**Comment by martinacantaro** [comment]
Created a new issue per @Jrigada 's request to track the remaining work not covered by this PR. https://github.com/lambdaclass/cairo-rs/issues/736

This one is now ready for final review!

**Pull request merged**